components/core/src/os/system/linux.rs

```diff
@@ -1,17 +1,15 @@
-use std::{ffi::CStr,
-          mem};
-
-use libc;
-
 use crate::{error::{Error,
                     Result},
             os::system::Uname};
 use errno::errno;
+use libc;
+use std::{ffi::CStr,
+          mem};
 
 pub fn uname() -> Result<Uname> { unsafe { uname_libc() } }
 
 unsafe fn uname_libc() -> Result<Uname> {
-    let mut utsname: libc::utsname = mem::uninitialized();
+    let mut utsname: libc::utsname = mem::MaybeUninit::uninit().assume_init();
```

Comment:
I believe this is equivalent to using `mem::uninitialized()` directly and still has the potential for undefined behavior. To avoid undefined behavior we need to initialize the memory before calling `assume_init`. So something like:

```rust
    let mut utsname = mem::MaybeUninit::uninit();
    let rv = libc::uname(utsname.as_mut_ptr());
    let utsname = utsname.assume_init();
```

[Here](https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#examples) are good examples of correct usage.

Reply:
Fixed in https://github.com/habitat-sh/habitat/pull/7177/commits/9f7d4a529e190195ebb7f23238d3942616ff68b6